### PR TITLE
Instagram Photos of Feed on Map. [Work in Progress]

### PR DESCRIPTION
**Pull request by prasht63**
Fixes https://github.com/loklak/loklak_webclient/issues/293
image
- [x] Enable oAuth to connect Accounts
- [x] Show photos from feed on Instagram.
- [x] Open Photos on PopUp
- [x] Show Profile Info of the Photographer
- [x] Show Hashtags
- [x] Show description

### CODE REVIEW

**Comment by prasht63** [comment]
@fuksi Please review and merge.Currently its not possible to store the Access Token in Loklak, so the URL shows the access token, once It is supported I will add the functionality to retrieve it from Loklak_Server

TestLink : http://115.99.198.192:3000/

Method to test : 
Go to /connect
Click on Add App for Instagram
Map should show images from your feed

**Comment by mariobehling** [comment]
Cannot login on test site currently.

**Comment by prasht63** [comment]
@mariobehling Can you try again. Same link. I have fixed the setup. Also even if you are not able to login.Just goto https://instagram.com/oauth/authorize/?client_id=a98ae74120464cf6a92f63ee9f4d90e9&redirect_uri=http://115.99.198.192:3001/instagram&response_type=token

**Comment by mariobehling** [comment]
Authenticating with Angel.co does not work. It wants to connect to localhost: http://localhost:3000/startups?code=2415f56b295ea81a70160726f1aadfd42d05d2e7c3a134de

**Comment by fuksi** [comment]
- [ ] The initial position of the map is strange
image
- [ ] It's maybe not the best idea to expose access_token in address bar :) If it's too hard to change the protocol, I think you can just hash it and show something else 
image

There can be of course improvement in pop up style but that's for later. Let's focus on getting it up & running :+1:

**Comment by prasht63** [comment]
@mariobehling This is for Instagram, Angel.co is another PR
@fuksi I mentioned problem 1 here https://github.com/loklak/loklak_webclient/pull/518#issuecomment-134747250 . For  now I will hash it.
About strange Map position I will see.

**Comment by mariobehling** [comment]
@prasht63 Are you still working on this issue?

File: app/js/controllers/insta.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      var i=0;
+      getData(url,1);
+      
+
+    
+
+
+  $scope.curr_page=0;
+  $scope.lastpage;
+     function getData(url,i)
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: ['getData' was used before it was defined.](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l53)

File: app/js/controllers/insta.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+         {
+            console.log(response);
+         });
+
+    }
+      
+
+   function plotOnMap()
+   {
+    console.log("called");
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l89)

File: app/js/controllers/insta.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
+'use strict';
+/* global angular, L */
+/* jshint unused:false */
+
+var controllersModule = require('./_index');
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l5)

File: app/js/controllers/insta.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        plotOnMap();
+        if(i<20)
+        {
+          i++;
+         url=url+"&callback=JSON_CALLBACK";
+         getData(url,i);
+        }
+      }, function(response)
+         {
+            console.log(response);
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l81)

File: app/js/controllers/insta.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                    for (i = 0; i < result.features.length; i++) {
+                      console.log(i);
+                        //console.log(result.features[i].propic-url);
+                        var tweetIcon = L.icon({
+                        iconUrl: result.features[i].properties.propic ,
+                        iconSize:     [30, 30]
+                    });
+                        var lat = result.features[i].geometry.coordinates[0];
+                        var lng = result.features[i].geometry.coordinates[1];
+                        console.log(lat+" , "+lng+" plotted");
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l132)

File: app/js/controllers/insta.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        '<a href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, ' +
+        'Imagery © <a href="http://mapbox.com">Mapbox</a>' +
+                'Imagery © <a href="http://mapbox.com">Mapbox</a>',
+            id: 'examples.map-20v6611k'
+        });
+        var basemapObj = {
+            "First Basemap": grayscale
+        };
+      var url="https://api.instagram.com/v1/users/self/feed?callback=JSON_CALLBACK&count=20&access_token=";
+      var map2 =L.map('map2',{layers:[grayscale,photoslayer]}).setView([20,0], 2);
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l37)

File: app/js/controllers/insta.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+      var access_token;
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: [Identifier 'access_token' is not in camel case.](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l38)

File: app/js/controllers/insta.js [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+      var query = window.location.href;
+      var vars = query.split("#");
+      var pairs=vars[1].split("=");
+      access_token=pairs[1];
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: [Identifier 'access_token' is not in camel case.](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l42)

File: app/js/controllers/insta.js [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+      url=url+access_token;
+      var i=0;
+      getData(url,1);
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: ["getData" was used before it was defined](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l45)

File: app/js/controllers/insta.js [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+            "First Basemap": grayscale
+        };
+      var url="https://api.instagram.com/v1/users/self/feed?callback=JSON_CALLBACK&count=20&access_token=";
+      var map2 =L.map('map2',{layers:[grayscale,photoslayer]}).setView([20,0], 2);
+      var access_token;
+      var query = window.location.href;
+      var vars = query.split("#");
+      var pairs=vars[1].split("=");
+      access_token=pairs[1];
+      url=url+access_token;
```
Comment (sudheesh001) [inline app/js/controllers/insta.js]: Codacy Issue found: [Identifier 'access_token' is not in camel case.](https://www.codacy.com/app/sudheesh1995/loklak_webclient/file/2550237812/issues/source?bid=3288768&fileBranchId=3288807#l43)

**Pull request closed without merging**